PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py

```diff
@@ -212,9 +220,15 @@ def __call__(self, process,
         self.setParameter('postfix',postfix),
 
         #if mva/puppi MET, autoswitch to std jets
-        if metType == "MVA" or 'Puppi' in postfix:
+        if metType == "MVA" or metType == "Puppi":
             self.setParameter('CHS',False),
 
+        #enabling puppi flag
+        self.setParameter('Puppi',self._defaultParameters['Puppi'].value) 
+        if metType == "Puppi":
+            self.setParameter('metType',"PF") 
+            self.setParameter('Puppi',True) 
```

Comment:
Can you just assume that users can pass this new "Puppi" parameter directly?
The override of metType back to PF looks confusing.
The documentation says that metType can be PF and Puppi, but whatever is passed to the constructor, the end result remains unchanged "PF".